File: 734/734b.cpp
#include <iostream>
using namespace std;
int main () {
	long long int a, b, c, d; cin >> a >> b >> c >> d;
	long long int m = 0;
	long long int ans = 0;
	a < c ? m = a : m = c;
	m > d ? m = d : m = m;

	a -= m; c -= m; d -=m;
	ans += m * 256;

	m = min(a, b);
	ans += m * 32;

	cout << ans << endl;
}

File: 734/734b.py
a = long(input())
b = long(input())
c = long(input())
d = long(input())

m = min(a, c, d)
count = 0
a = a - m; c = c - m; d = d - m
count = m * 256

m = min(a, b)
count = count + m * 32

print(str(count))